### PR TITLE
Create 0701-insert-into-a-binary-search-tree.js

### PR DESCRIPTION
**Pull request by aadil42**
Solved insert-into-a-binary-search-tree in JS.

File(s) Added: 0701-insert-into-a-binary-search-tree.js
Language(s) Used: JavaScript
Submission URL: https://leetcode.com/problems/insert-into-a-binary-search-tree/submissions/1354476551/

### CODE REVIEW

File: javascript/0701-insert-into-a-binary-search-tree.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ */
+var insertIntoBST = function(root, val) {
+    
+    const dfs = (root) => {
+        if(!root) {
+            return new TreeNode(val);
+        }
+        if(val > root.val) {
+            root.right = dfs(root.right);
+        }  else {
```
[the remark this answers is not in the document]
Comment (aadil42) [reply javascript/0701-insert-into-a-binary-search-tree.js:24]: `else` omitted.
Submission for the proposed code: https://leetcode.com/problems/insert-into-a-binary-search-tree/submissions/1388682788/

```javascript
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * h = height of the tree, could be n.
 * Time O(h) | Space O(h)
 * @param {TreeNode} root
 * @param {number} val
 * @return {TreeNode}
 */
var insertIntoBST = function(root, val) {
    return dfs(root, val);
};

const dfs = (root, val) => {
    if(!root) {
        return new TreeNode(val);
    }
    if(val > root.val) {
        root.right = dfs(root.right, val);
        return root;
    }
    root.left = dfs(root.left, val);
    return root;
}

```

[the remark this answers is not in the document]
Comment (aadil42) [reply javascript/0701-insert-into-a-binary-search-tree.js:24]: @aakhtar3 (tagging to draw attention to the proposed code)

[the remark this answers is not in the document]
Comment (Ykhan799) [reply javascript/0701-insert-into-a-binary-search-tree.js:24]: @aadil42 Proposed Solution looks good.
1. Add Spaces to control flow statements.
2. Update PR file reflecting your proposed code

[the remark this answers is not in the document]
Comment (aadil42) [reply javascript/0701-insert-into-a-binary-search-tree.js]: @Ykhan799, It's done. Please review.

**Comment by Ykhan799** [comment]
@aadil42 thank you for applying my suggestions, I merged all the PRs I reviewed

**Pull request merged**